### DISCUSSION TITLE
Migration strategy brainstorm

### DISCUSSION THREAD

**Discussion by LordTermor** [anydistro/bxt#20] [General]
Since boxit and bxt are fully incompatible between each other, we need to find an optimal migration strategy.

Some ideas:

---

### Parallel running
Having a dedicated testing instance of bxt so we can teach maintainers to operate it, find bugs and get feedback

#### Pros:
- Safe - no need to change our current repositories in any way
#### Cons:
- Hard for maintainers as it doubles the work
---
### Incremental migration
Migrate only unstable branch to bxt while keeping testing and stable on boxit

#### Pros:
- A bit easier for maintainers
- Still can find bugs without affecting stable/testing users
#### Cons:
- Would be _really_ tricky to implement, maybe changes to boxit are needed
- A chance blow up the unstable branch completely
- We also need to think about the system rollback if something goes wrong

**Comment by hphilm** [comment]
We go for parallel running. Those maintainers with boxit access should continue to push their packages there. Some maintainers who want to test BXT add additional packages to BXT for testing the system. BXT will point to Boxit server as upstream source. This way packages pushed to boxit will be pulled by BXT also. BXT need a set of test packages to test the process. The packages can be dummy packages or AUR packages normally not found on boxit.

When we tested all functions of BXT we can create a migration plan to see how we move from boxit to BXT.

* keep the old overlay and sync pool structure to not blow up mirrors
* any change in structure, except symlinks may cause downtime or mirrors not working/syncing
* migration needs to be tested on some live systems before we do the switch

**Reply by hphilm** [reply]
@oberon-manjaro should start as soon as possible to test list builds. Kernel builds might be special and might need extra features if an incomplete set got built. There we need to email the kernel maintainers and hold back the packages to been published. I recommend a staging area on BXT server which needs manual intervention to let them get published to a branch. For the beginning we can keep those failed sets on github, which then need to been uploaded manually.

**Comment by fhdk** [comment]
I'd like to offer my assistance anywhere possible perhaps

* python scripting for interacting with the backend
* testing
* ?
